.github/workflows/CI.yml

```diff
@@ -234,134 +241,32 @@ jobs:
-        uses: docker/setup-buildx-action@v2
-        id: buildx
-        with:
-          install: true
-
-      - name: Login to Docker Hub
-        uses: docker/login-action@v2
-        # only run docker login on pushes; also for PRs, but only if this is not a fork
-        if: ( github.actor != 'renovate[bot]' && github.actor != 'dependabot[bot]' ) && (matrix.config.should-push-image == 'true' && (github.event_name == 'push' || github.event_name == 'workflow_dispatch' || github.event.pull_request.head.repo.full_name == github.repository))
-        # note: GH does not allow to access secrets for PRs from a forked repositories due to security reasons
-        # that's fine, but it means we can't push images to dockerhub
-        with:
-          username: ${{ secrets.REGISTRY_USER }}
-          password: ${{ secrets.REGISTRY_PASSWORD }}
-
-      - id: docker_build_image
-        name: "Docker Build keptn/${{ matrix.config.artifact }}"
-        if: matrix.config.should-push-image == 'true' && ( matrix.config.should-run == 'true' || needs.prepare_ci_run.outputs.BUILD_EVERYTHING == 'true' )
-        uses: docker/build-push-action@v3
-        with:
-          context: ${{ matrix.config.working-dir }}
-          tags: |
-            keptndev/${{ matrix.config.artifact }}:${{ env.VERSION }}
-            keptndev/${{ matrix.config.artifact }}:${{ env.VERSION }}.${{ env.DATETIME }}
-          build-args: |
-            version=${{ env.VERSION }}
-            buildTime=${{ needs.prepare_ci_run.outputs.DATETIME }}
-            gitSha=${{ needs.prepare_ci_run.outputs.GIT_SHA }}
-          push: ${{ matrix.config.should-push-image == 'true' && (( github.event_name == 'pull_request' && github.event.pull_request.head.repo.full_name == github.repository ) || ( github.event_name == 'push' )) && github.actor != 'renovate[bot]' && github.actor != 'dependabot[bot]' }}
-          pull: true
-          builder: ${{ steps.buildx.outputs.name }}
-          cache-from: type=gha, scope=${{ github.workflow }}
-          cache-to: type=gha, scope=${{ github.workflow }}
+    uses: ./.github/workflows/test-and-build-docker-images.yml
+    secrets: inherit
+    with:
+      artifact: ${{ matrix.config.artifact }}
+      tags: |
+        keptndev/${{ matrix.config.artifact }}:${{ needs.prepare_ci_run.outputs.VERSION }}
+        keptndev/${{ matrix.config.artifact }}:${{ needs.prepare_ci_run.outputs.VERSION }}.${{ needs.prepare_ci_run.outputs.DATE }}${{ needs.prepare_ci_run.outputs.TIME }}
+      should-run: ${{ matrix.config.should-run }}
+      should-push-image: ${{ matrix.config.should-push-image == 'true' && (( github.event_name == 'pull_request' && github.event.pull_request.head.repo.full_name == github.repository ) || ( github.event_name == 'push' )) && github.actor != 'renovate[bot]' && github.actor != 'dependabot[bot]' }}
+      docker-test-target: ${{ matrix.config.docker-test-target }}
+      working-dir: ${{ matrix.config.working-dir }}
```

Comment:
why are you passing the matrix values through separately when you also define the matrix above as well?
is it not enough to just define the matrix?

Reply:
I used this approach for future reusability,  so you wouldn't need a matrix to run this workflow.
In this case the matrix strategy is applied on the invoking workflow stage.
The alternative would be to pass the whole matrix into the reusable workflow and apply the strategy there.